jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -497,11 +484,15 @@ def mos_niriss_parser(app, data_dir, obs_label=""):
 
         image_label = "Image {} {}".format(im_split[0], pupil)
 
-        image_data = _parse_as_image(image_file)
+        with fits.open(image_file) as file_obj:
+            data_iter = get_image_data_iterator(app, file_obj, "Image", ext=None)
+            data_obj = [d[0] for d in data_iter]  # We do not use the generated labels
+            image_data = data_obj[0]  # Grab the first one. TODO: Error if multiple found?
```

Comment:
Imviz parser can take multiple extensions, so maybe this patch also solves the problem of not able to load image with multiple extensions in Mosviz (Brian Cherinka said that is a problem, right? I cannot keep track anymore). Though right now, I am only grabbing the first one, so more work still needs to be done on that front but at least it should no longer crash.

Reply:
@havok2063 , does this help with the problem you were describing?